packages/relay/src/lib/services/metricService/metricService.ts

```diff
@@ -102,19 +138,23 @@ export default class MetricService {
    */
   constructor(
     logger: Logger,
-    sdkClient: SDKClient,
+    hapiService: HAPIService,
```

Comment:
QQ: Why was this dependency changed? 

It seems that we are using it only to get the `SDKClient`:

```typescript
this.sdkClient = hapiService.getSDKClient();
```

Reply:
To me it made more sense to pass in the `HAPIService` into the `MetricService` since the `HAPIService` is responsible for resetting the client.   The client functionality as a whole is better represented in the `HAPIService` IMHO.  If it is too confusing I can set it back.

I'll answer your general question here since I cannot find a way to respond to it directly.  As stated in the `Description` , "Labels Are Not Meant for Numerical Data".  That's what the original implementation was doing.  This made visualizing the metrics in grafana challenging.  There are three conditions in which the client will be reset in the service:
1. A number of transactions processed
2. Predefined error codes being returned from the consensus node
3. The duration since the last client reset expiring.

The goal is to understand how often and why the client gets reset, in order to better understand the causes of any failed transactions.